### ISSUE TITLE
Zeit.Now - Next build crashes when using "env" in next.config.js instead of publicRuntimeConfig

### ISSUE THREAD

**Issue by Vadorequest** [zeit/next.js#8907]
# Bug report

## Describe the bug

I'm using Zeit.Now and I was using `publicRuntimeConfig` in `next.config.js` just fine until I deployed, which yield `Cannot use publicRuntimeConfig with target=serverless`. The reasons is [there](https://github.com/zeit/next.js#build-time-configuration) but when I replace my `publicRuntimeConfig` with `env`, the process hangs indefinitely with the cryptic error `at getBaseWebpackConfig (/Users/vadorequest/dev/TFP/node_modules/next/dist/build/webpack-config.js:91:95)`

## To Reproduce

Replace `publicRuntimeConfig` with `env` in `next.config.js` and run `now dev`

## Expected behavior

Should build properly like https://github.com/zeit/next.js#build-time-configuration states.

## Screenshots

![image](https://user-images.githubusercontent.com/3807458/65841266-da27a280-e320-11e9-9f71-56c530b16cb2.png)

## System information

- OS: [e.g. macOS, Windows] MacOS
- Version of Next.js: 9.0.7

## Additional info

Here is my `next.config.js`

```js

require('dotenv').config({ path: '.env' });

console.debug(`Building Next with NODE_ENV="${process.env.NODE_ENV}"`);

module.exports = (phase, { defaultConfig }) => {
  return {
    env: { // See https://github.com/zeit/next.js#build-time-configuration
      ORGANISATION_NAME: process.env.ORGANISATION_NAME,
      NODE_ENV: process.env.NODE_ENV, // XXX Used in utils/env
      GRAPHCMS_CACHE_ENDPOINT: process.env.GRAPHCMS_CACHE_ENDPOINT,
    },
    webpack: config => {
      // Fixes npm packages that depend on `fs` module
      config.node = {
        fs: 'empty',
      };

      return config;
    },
  };
};
```

**Comment by Vadorequest** [comment]
I got a more detailed stack trace by doing the following:

```
 ↵ 1  yarn dev
yarn run v1.17.3
$ next dev --port 8888
[ wait ]  starting the development server ...
[ info ]  waiting on http://localhost:8888 ...
Building Next with NODE_ENV="development"
Error: The key "NODE_ENV" under "env" in next.config.js is not allowed. https://err.sh/zeit/next.js/env-key-not-allowed
    at webpackConfig.plugins._webpack.default.DefinePlugin.Object.keys.reduce (/Users/vadorequest/dev/TFP/node_modules/next/dist/build/webpack-config.js:91:161)
    at Array.reduce (<anonymous>)
    at getBaseWebpackConfig (/Users/vadorequest/dev/TFP/node_modules/next/dist/build/webpack-config.js:91:95)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

**Comment by Vadorequest** [comment]
Removing `NODE_ENV: process.env.NODE_ENV` from the file did fix the issue. Launching with the beta `now dev` was hiding a self-explanatory error.

**Comment by leo** [comment]
Thanks for reporting!

It sucks that `now dev` didn't properly render the error and instead hid it away. As we speak, my team is working on a more refined version of the output, which simply relays exactly what your development tool is printing, without hiding lines away.

I'll let you know once it has launched!

**Comment by Vadorequest** [comment]
@leo  I guess we should close this? I don't know if the CLI was improved since then.

**Comment by leo** [comment]
Yep! The error is now being surfaced. Thank you for reporting!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
